notebooks/217-vision-deblur/217-vision-deblur.ipynb

```diff
@@ -0,0 +1,424 @@
+   "id": "a15f4c33-d7fc-47a4-9e2b-9a1c94a51dde",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_downloaded_dir = (\n",
+    "    base_model_dir / Path(model_info_json[\"subdirectory\"]) / Path(precision)\n",
+    ")\n",
+    "\n",
+    "for model_file in model_downloaded_dir.iterdir():\n",
+    "    try:\n",
+    "        shutil.copyfile(model_file, model_dir / model_file.name)\n",
+    "    except FileExistsError:\n",
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d0a38ab9-fea0-434d-b855-113799b51011",
+   "metadata": {},
+   "source": [
+    "## Load the Model\n",
+    "\n",
+    "Load and compile the DeblurGAN-v2 model in the Inference Engine with `ie.read_model` and compile it for the specified device with `ie.compile_model`. Get input and output keys and the expected input shape for the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "baea1ef6-08fa-48e6-bac9-56e57eede3c4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ie = Core()\n",
+    "model = ie.read_model(model=model_xml_path)\n",
+    "compiled_model = ie.compile_model(model=model, device_name=DEVICE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9b260f8b-9034-4dcc-b894-472125bf542c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_input_layer = compiled_model.input(0)\n",
+    "model_output_layer = compiled_model.output(0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "86f845ef-1be3-4b22-9f75-23a5ec985017",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_input_layer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b0bad1b4-c0ac-4ad3-aeda-2e2035a52c21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_output_layer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00f1d1c5-64e8-432e-b90a-6dac37d643ee",
+   "metadata": {},
+   "source": [
+    "## Deblur Image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c0861896-7579-461c-80de-720833616b68",
+   "metadata": {},
+   "source": [
+    "### Load, resize and reshape input image\n",
+    "\n",
+    "The input image is read using the default `load_image` function from notebooks.utils, resized to meet the network's expected input sizes, and reshaped to (N, C, H, W), where N=number of images in the batch, C=number of channels, H=height, and W=width."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e1b4718a-8711-4bcd-a382-08e88c227d0c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Image filename\n",
+    "filename = \"000144_19.png\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "af40f278-06a7-4f70-9217-372097c250f2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load input image\n",
+    "image = load_image(f\"data/blur/{filename}\")\n",
```

Comment:
We should avoid saving data into the repo. Could we use a URL instead of saving the images into the repo?

Reply:
Hello. The relative path you mention points to an image sample of the GOPRO dataset, which is only available in a .zip format online.  Do you want me to create a URL for the specific image, or simply restructure the code and delete the corresponding data from the repo?

Reply:
Yes, please create a URL for the specific image you used in your notebook. Please also restructure the code to read the image from the URL, instead of reading it from the 'data' folder. The data folder should be removed from the repo.